pages/draft/2020-12/migration-notes.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Draft 2019-09 to Draft 2020-12
+section: docs
+---
+
+### Introduction
+
+JSON Schema Draft 2020-12 introduces key improvements over Draft 2019-09, focusing on improved modularity and cross-referencing capabilities. This means improvements for schema resuse, being able to define features a schema supports, etc. This guide covers all major changes and how to migrate from Draft 2019-09 to Draft 2020-12.
+
+### Keyword changelog
+
+| Keyword (Draft 2019-09) | Keyword (Draft 2020-12) | Specification | Keyword Type | Behavior Details |
+| --- | --- | --- | --- | --- |
+| $anchor | remained $anchor | Core |  | In 2019-09, $anchor could use a colon and not start with an underscore, but in 2020-12, the upgrade in syntax rule can't use a colon. |
+| $schema | remained $schema | Core |  | The behavior was updated to use dialect 2020-12  |
+| $recursiveAnchor | `$dynamicAnchor` | Core | Identifier | Similar to `$anchor`, but supports `$dynamicRef` for flexible schema resolution and $dynamicAnchor now has its own syntax. |
+| $recursiveRef | `$dynamicRef` | Core | Identifier | Works with `$dynamicAnchor` to allow referencing dynamically resolved schemas at runtime. |
+| $vocabulary | `$vocabulary` | Core | Identifier | Specifies which vocabularies a schema uses in 2019-09, but the update now uses format-annotation aiding compatibility and extensibility. |
+| `items` (used for both lists and tuples) | `prefixItems` | Core | Applicator/Annotation | Separates tuple validation (`prefixItems`) from list validation (`items`). Now, `items` applies only to additional elements beyond `prefixItems`. |
+| $vocabulary |  | Core |  |  |
+
+### Tutorial
+
+
+#### Migration Steps
+
+##### 1. Updating `$schema` Declaration
+
+Reference the correct Draft 2020-12 meta-schema to ensure compatibility:
+
+**Before (Draft 2019-09):**
+
+```
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema"
+}
+```
+
+**After (Draft 2020-12):**
+
+```
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema"
+}
+```
+
+##### 2. Using `$dynamicAnchor` and `$dynamicRef`
+
+Switch to `$dynamicAnchor` when defining flexible anchors:
+
+**Before (Draft 2019-09):**
+
+```
+{
+  "$anchor": "mySchema"
+}
+```
+
+**After (Draft 2020-12):**
+
+```
+{
+  "$dynamicAnchor": "mySchema"
+}
+```
```

Comment:
`$dynamicAnchor` replaces `$recursiveAnchor`, not `$anchor`.

`````suggestion
Switch to `$dynamicAnchor` when defining recursive schemas:

**Before (Draft 2019-09):**

```
{
  "$recursiveAnchor": true
}
```

**After (Draft 2020-12):**

```
{
  "$dynamicAnchor": "mySchema"
}
```
`````